### ISSUE TITLE
Figure out how to route requests to the sftp service

### ISSUE THREAD

**Issue by slifty** [PermanentOrg/infrastructure#108]
Right now the SFTP service gets deployed but we will want port 22 requests to get routed to it (e.g. dev.permanent.org:22 should not go to the backend but rather to the sftp service).

Alternatively we could do something like `sftp.dev.permanent.org`.

Either way, we need to set up appropriate DNS routing.

**Comment by slifty** [comment]
Did some research around this and here's what I've learned...

1. For each deployment (e.g. dev, staging, prod) we have DNS entries that point to an Application Load Balancer in AWS.

2. The load balancer routes various port traffic to two different Target Groups (one for the backend, one for the upload service).

3. What we're proposing is adding a new service on port 22.

Unfortunately, AWS  Application Load Balancers *ONLY* support HTTP.  This means we cannot simply update the existing balancers with a new rule.  This means we have two options:

* Replace the "Application Load Balancer" with a new "Network Load Balancer" which does load balancing for both HTTP and SSH traffic.  Network Load Balancers exist lower in the OSI model [and are not aware of application context for things like routing and health checks](https://medium.com/awesome-cloud/aws-difference-between-application-load-balancer-and-network-load-balancer-cb8b6cd296a4).

* Create a new "Network Load Balancer" with a new DNS record (e.g. `sftp.dev.permanent.org`) which is used for non-http based load balancing.

**Comment by cecilia-donnelly** [comment]
We talked about this in our sprint meeting yesterday. We use some specific rules in our "Application Load Balancer" that we want to keep, so we discussed putting a "Network Load Balancer" *in front of* the ALB. So requests would come in, be routed by the NLB depending on protocol, then if HTTP/HTTPS go to the ALB as they do currently.

This [AWS blog post](https://aws.amazon.com/blogs/networking-and-content-delivery/application-load-balancer-type-target-group-for-network-load-balancer/) explains that we can do this by setting up the ALB as a target group for the NLB to point to.
